### PR TITLE
Third party authorised room invites

### PR DESCRIPTION
**Pull request by Cadair**
This is something I have had rattling around in my head for a while. If there's enough positive feedback from the community and someone else helps me mentor it, I think it would be a cool project.

### CODE REVIEW

File: _projects/authenticated_3rd_party_invites.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
+name: 3rd Party Authorised Invites
+desc: Invite your users to a room based on some external authorisation. 
+# add a short one line description of your project
+requirements:
+ - Understanding of the matrix Client-Server API
+difficulty: medium
+issues:
+mentors:
+ - Cadair
```
[the remark this answers is not in the document]
Comment (Half-Shot) [reply _projects/authenticated_3rd_party_invites.md:10]: Yeah, put me down :)
```suggestion
 - Cadair
 - Half-Shot
```

File: _projects/authenticated_3rd_party_invites.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ - Familiarity with a programming language with a good matrix SDK (for the
+   server side component).
+difficulty: medium
+issues:
+mentors:
+ - Cadair
+ - Half-Shot
+initiatives:
+ - GSoC
+tags:
```
Comment (thibaultamartin) [inline _projects/authenticated_3rd_party_invites.md:16]: Some of the pure tech requirements from `requirements` should probably be moved here. I'd leave "Understanding of the matrix Client-Server API" under `requirements` though

File: _projects/authenticated_3rd_party_invites.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+---
+
+#### Description
+
+This project aims to fill a gap which currently exists in the matrix
+ecosystem, which is the ability to delegate membership of a room to a third
+party service.
+
+Two examples of this workflow include:
```
Comment (thibaultamartin) [inline _projects/authenticated_3rd_party_invites.md:25]: Three :)

File: _projects/authenticated_3rd_party_invites.md (hunk 2 above)
Comment (thibaultamartin) [inline _projects/authenticated_3rd_party_invites.md:16]: Can you also add either a `175h` or `350h` tag depending on how long you think the project should take?

File: _projects/authenticated_3rd_party_invites.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+1. A (Patreon) supporters room: if you support a project at a certain level you
```
Comment (thibaultamartin) [inline _projects/authenticated_3rd_party_invites.md:27]: One thing to keep in mind in the Patreon case: Patreon doesn't give people's MxID, but their email address. I don't think it's possible to add people to the Patreon supporters room directly without emailing people a link to a form so they give their MxID

**Review by thibaultamartin — COMMENTED** [review]
Thanks, that looks like an exciting project and I'd love to see it happen!

File: _projects/authenticated_3rd_party_invites.md (hunk 4 above)
Comment (Cadair) [reply _projects/authenticated_3rd_party_invites.md:27]: The example workflow is that you click a link in the Patreon, login to your matrix account and then the bot has a verified link between your Patreon account and your matrix id. Meaning the bot can auth you for invites and check up in the future on your Patreon status.

File: _projects/authenticated_3rd_party_invites.md (hunk 2 above)
Comment (Cadair) [reply _projects/authenticated_3rd_party_invites.md:16]: I don't think the element knocking one had this either?

File: _projects/authenticated_3rd_party_invites.md (hunk 4 above)
Comment (Cadair) [reply _projects/authenticated_3rd_party_invites.md:32]: I added a sentence about this to hopefully make it clearer.

**Review by Half-Shot — APPROVED** [review]
Proposal looks sound 👍

File: _projects/authenticated_3rd_party_invites.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+* A Javascript frontend which logs into the matrix account and gets an OpenID
+  token to pass to the backend. Using OpenID like this allows the server to
+  verify the matrix id of the invitee while not having an access token for the
+  users matrix account.
+* A plugin system allowing the server admin to configure what third party
+  systems to use for authorisation.
+
+Other extensions to this project could include:
+
+* Perodic checks against the third party service to kick people from the rooms
```
[the remark this answers is not in the document]
Comment (Cadair) [reply _projects/authenticated_3rd_party_invites.md:59]: Wanna move it to the main part of the text?

[the remark this answers is not in the document]
Comment (Cadair) [reply _projects/authenticated_3rd_party_invites.md]: have moved it.

**Pull request merged**